admin/controllers/results.js

```diff
@@ -36,10 +36,12 @@ controller.getViewResultsPage = async (req, res, next) => {
     return next(error)
   }
   const currentDate = moment.tz(req.user.timezone || config.DEFAULT_TIMEZONE)
-  const isResultsPageAccessible = schoolHomeFeatureEligibilityPresenter.isResultsPageAccessible(currentDate, checkWindow)
-  const nationalScore = resultPresenter.getScoreWithOneDecimalPlace(checkWindow.score)
-  schoolScore = resultPresenter.getScoreWithOneDecimalPlace(schoolScoreRecord && schoolScoreRecord.score)
-  if (!isHdfSubmitted || !isResultsPageAccessible) {
+
+  const isResultsPageAccessibleForSubmittedHdfs =
+    schoolHomeFeatureEligibilityPresenter.isResultsPageAccessibleForSubmittedHdfs(currentDate, checkWindow, isHdfSubmitted)
```

Comment:
Given that optimisation is now in full swing, how would you cache this to remove the need to query the db in the first place?

Reply:
@jon-shipley sounds like this along with the rest of the features can be revisited once redis cache pt.1 is in play?